### DISCUSSION TITLE
Clerk with react-router-dom

### DISCUSSION THREAD

**Discussion by thewhalermelon** [clerk/javascript#2920] [Support request - DO NOT OPEN HERE]
Dear Clerk Support Team,

I am developing a project using ReactJS and have integrated Clerk into my application for managing user authentication processes, including email verification functionality. I am using react-router-dom to manage routing within my React application.

I am encountering an issue after user signup: Clerk automatically redirects users to the /sign-up/verify-email-address route so they can enter the verification code sent via email. However, I've noticed that if I don't declare this route in react-router-dom, the page displays as a blank page or redirects back to the default page if no corresponding route exists.

I would like to know how to correctly declare the /sign-up/verify-email-address route in react-router-dom and what the default Clerk component for this page is, so users can smoothly enter their email verification code.

Do I need to provide a custom component to handle this page? Or does Clerk offer an integrated solution that I can utilize?

Any assistance or guidance from you would be greatly appreciated. I aim to provide a seamless and smooth authentication experience for my users.

Thank you and best regards,

**Comment by emersonh-xyz** [comment]
Had a similar issue and was able to resolve here
https://github.com/orgs/clerk/discussions/2900#discussioncomment-8723836
